src/Uno.UI.Runtime.Skia.X11/X11NativeElementHostingExtension.cs

```diff
@@ -0,0 +1,285 @@
+	{
+		if (content is not X11NativeWindow nativeWindow)
+		{
+			return false;
+		}
+
+		using var _1 = X11Helper.XLock(_display);
+
+		var _3 = XLib.XQueryTree(_display, XLib.XDefaultRootWindow(_display), out IntPtr root, out _, out var children, out _);
+		var _4 = XLib.XFree(children);
+
+		// _NET_CLIENT_LIST only identifies top-level windows, not subwindows.
+		var status = XLib.XGetWindowProperty(
+			_display,
+			root,
+			X11Helper.GetAtom(_display, X11Helper._NET_CLIENT_LIST),
+			0,
+			new IntPtr(0x7fffffff),
+			false,
+			X11Helper.AnyPropertyType,
+			out _,
+			out _,
+			out var length,
+			out _,
+			out IntPtr windowArray);
+
+		if (status == X11Helper.Success)
+		{
+			unsafe
+			{
+				var span = new Span<IntPtr>(windowArray.ToPointer(), (int)length);
+				foreach (var window in span)
+				{
+					if (window == nativeWindow.WindowId)
+					{
+						return true;
+					}
+				}
+			}
+		}
+
+		return FindWindowById(_display, nativeWindow.WindowId, root) != IntPtr.Zero;
+	}
+	public void AttachNativeElement(object content)
+	{
+		if (content is X11NativeWindow nativeWindow
+			&& XamlRoot is { } xamlRoot
+			&& X11Manager.XamlRootMap.GetHostForRoot(xamlRoot) is X11XamlRootHost host)
+		{
+			using var _1 = X11Helper.XLock(_display);
+
+			// this seems to be necessary or else the WM will keep detaching the subwindow
+			XWindowAttributes attributes = default;
+			var _2 = XLib.XGetWindowAttributes(_display, nativeWindow.WindowId, ref attributes);
+			attributes.override_direct = /* True */ 1;
+
+			unsafe
+			{
+				IntPtr attr = Marshal.AllocHGlobal(Marshal.SizeOf(attributes));
+				Marshal.StructureToPtr(attributes, attr, false);
+				var _3 = X11Helper.XChangeWindowAttributes(_display, nativeWindow.WindowId, (IntPtr)XCreateWindowFlags.CWOverrideRedirect, (XSetWindowAttributes*)attr.ToPointer());
+				Marshal.FreeHGlobal(attr);
+			}
+
+			var _4 = X11Helper.XReparentWindow(_display, nativeWindow.WindowId, host.RootX11Window.Window, 0, 0);
+			XLib.XSync(_display, false); // XSync is necessary after XReparent for unknown reasons
+
+			using var _5 = X11Helper.XLock(_display);
+			var _6 = X11Helper.XRaiseWindow(host.TopX11Window.Display, host.TopX11Window.Window);
+
+			if (!_hostToNativeElementHosts.TryGetValue(host, out var set))
+			{
+				set = _hostToNativeElementHosts[host] = new HashSet<X11NativeElementHostingExtension>();
+			}
+			set.Add(this);
+
+			xamlRoot.InvalidateRender += UpdateLayout;
+		}
+		else
+		{
+			throw new InvalidOperationException($"{nameof(AttachNativeElement)} called in an invalid state.");
+		}
+	}
+
+	public void DetachNativeElement(object content)
+	{
+		if (content is X11NativeWindow nativeWindow
+			&& XamlRoot is { } xamlRoot
+			&& X11Manager.XamlRootMap.GetHostForRoot(xamlRoot) is X11XamlRootHost host)
+		{
+			using var _1 = X11Helper.XLock(_display);
+			var _2 = XLib.XQueryTree(_display, nativeWindow.WindowId, out IntPtr root, out _, out var children, out _);
```

Comment:
You should be able to use `_` directly, afaik. The only scenario scenario that needs `_1` or similar is with using. Still, it feels like the usings "fake discards" should be named to avoid having to renumber...